src/Shared/QueryStringEnumerable.cs

```diff
@@ -88,21 +89,20 @@ public ReadOnlyMemory<char> DecodeName()
         public ReadOnlyMemory<char> DecodeValue()
             => Decode(EncodedValue);
 
-        private static unsafe ReadOnlyMemory<char> Decode(ReadOnlyMemory<char> chars)
+        private static ReadOnlyMemory<char> Decode(ReadOnlyMemory<char> chars)
         {
             // If the value is short, it's cheap to check up front if it really needs decoding. If it doesn't,
             // then we can save some allocations.
-            if (chars.Length < 16 && chars.Span.IndexOfAny('%', '+') < 0)
+            ReadOnlySpan<char> source = chars.Span;
+            if (source.Length < 16 && source.IndexOfAny('%', '+') < 0)
```

Comment:
This (existing) comment seems misguided.
It's cheaper to do the check on e.g. 32 chars compared to 7.

I'd drop drop the length check and have this be just `!source.ContainsAny('%', '+')`

Reply:
I had this original thought too as you mention.

But then later I arrived at a slightly different observation:
- most of these query key/value pairs are likely to be small and less likely contain contain any escaped data
- OR on the other hand, when they are large, it is more likely to contain data that was escaped.

So, it can say: for large inputs it is more likely that unescaping is required anyway, then don't bother checking if it contains `%` or `+`.

I don't have any stats to back up this 'theory' nor any way to confirm if similar intention was there when originally implemented. So that said I am happy to remove the condition if generally agreed.

Reply:
At a minimum, we should switch to `ContainsAny`

Reply:
I will do that. Do you also agree to remove the 16 limit, and then I would do that too.

Reply:
I have updated the code as suggested:
- using `ContainsAny`
- removing the 16 char limit check for small string

I updated the PR, some these existing perf checks have become 1-3 ns slower (compared to the previous results), I have posted the latest results in the description of the PR.